Backend/microservices/authority/src/main/java/com/ms/authority/service/UserService.java

```diff
@@ -158,4 +158,30 @@ private UserDto convertToUserDto(User user) {
                 .collect(Collectors.toSet()));
         return userDto;
     }
+
+    public void deleteUser( User user){
+        tokenService.deleteTokenByUser(user);
+        userRepository.delete(user);
+    }
+
+    public User updateUserById(User user, RegistrationRequest request){
+        userRepository.findById(user.getId()).map(userUpdated -> {
+            if (request.getFirstName() != null){
+                user.setFirstName(request.getFirstName());
+            }
+            if (request.getLastName() != null){
+                user.setLastName(request.getLastName());
+            }
+            if (request.getEmail() !=null){
+                user.setEmail(request.getEmail());
+            }
+            if(request.getRoles() !=null){
+                user.setRoles(request.getRoles().stream().map(
```

Comment:
please create method for extracting set of roles from set of strings and use it here and in registration method